### DISCUSSION TITLE
LTSP LDAP Authentication results in PAM local configuration change dialog box to pop up during client boot

### DISCUSSION THREAD

**Discussion by pzerone** [ltsp/ltsp#916] [General] [answered]
When using LDAP authentication with LTSP on ubuntu 22.04, A TUI popup is shown during evey LTSP client boot mentioning that PAM configs have been modified locally:
![1717862092](https://github.com/ltsp/ltsp/assets/46400271/52ada2a4-c79c-4176-88d3-01ffe71e7b1f)

### Things noted
- In  stock ubuntu 22.04 LTS VirtualBox VM image, Pressing `No` will boot the system and allow an LDAP user to login, *but* the created home directory is not persisted and is not in the LTSP server.
- Pressing `Yes` Creates the home directory properly on the server and is mounted to the LTSP client, but the popup comes back on every reboot
- In debian 12 gnome with `/` as LTSP image, pressing `No` shows a Gray background with *possibly* a loading animation, but is stuck there infinitly
![1717862488](https://github.com/ltsp/ltsp/assets/46400271/f9e89088-504e-41b2-8e30-c6576096e9ed)
- Pressing `Yes` does nothing.

### Additional Info
- LTSP server was installed on both bare metal and a proxmox VM. both has the same issue.
- For ldap, the package installed for the client was `libpam-ldapd` and the following line was added to `/etc/pam.d/common-session`
```
session optional pam_mkhomedir.so skel=/etc/skel umask=077
```
But the issue persists with or without this line on the file
LDAP server configured properly and doing a `getent passwd <username>` returns the entry, on both LTSP server and LTSP client images on regular boot.

**Comment by alkisg** [comment, marked as answer]
This is a local misconfiguration; converting into a discussion.

Changes in PAM should be done with care, so that if you run `sudo pam-auth-update` on the server, you don't see the dialog.
The content until the `# end of pam-auth-update config` line is owned by pam-auth-update and should not be edited manually.

**Comment by pzerone** [comment]
Thanks for the reply. Indeed, this was a local mis configuration. The LDAP mkhomedir.so module need not be added to the LTSP client images, but only on the LTSP server. And pam-auth-update does it for me when i selected that option in the dialog box, instead of editing the file manually.
